cmd/traffic_cache_tool/CacheTool.cc

```diff
@@ -1322,6 +1324,47 @@ Get_Response(FilePath const &input_file_path)
   return zret;
 }
 
+void static scan_span(Span *span)
+{
+  for (auto strp : span->_stripes) {
+    strp->loadMeta();
+    strp->loadDir();
+    strp->walk_all_buckets();
+    CacheScan cs(strp);
+    cs.Scan();
+
+    // break; // to be removed
+  }
+}
+
+Errata
+Scan_Cache()
+{
+  Errata zret;
+  Cache cache;
+  std::vector<std::thread> threadPool;
+  if ((zret = cache.loadSpan(SpanFile))) {
+    cache.dumpSpans(Cache::SpanDumpDepth::SPAN);
+    for (auto sp : cache._spans) {
+      std::thread t(scan_span, sp);
+      threadPool.push_back(std::move(t)); // move constructor is necessary since std::thread is non copyable
+      // break; // to be removed
+    }
+    for (auto &th : threadPool)
+      th.join();
+    for (auto urlV : CacheScan::global_url_map) {
+      std::cout << "\n++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
+                << "stripe: " << urlV.first << "\n"
+                << "total number of entries: " << CacheScan::global_url_map[urlV.first].size()
+                << "\n+++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n"
+                << CacheScan::global_url_map.size() << std::endl;
```

Comment:
Wow, that might be a REALLY BIG map...

Reply:
yah.. I had this map to store the urls if we want to write in a file or somethings. Not sure if it is worth storing though